tools/SwiftSourceKitClient/SourceKitdResponse.swift

```diff
@@ -0,0 +1,306 @@
+      return Int(count)
+    }
+
+    public init(arr: sourcekitd_variant_t) {
+      assert(sourcekitd_variant_get_type(arr).rawValue ==
+          SOURCEKITD_VARIANT_TYPE_ARRAY.rawValue)
+      self.arr = arr
+    }
+
+    public func getString(_ index: Int) -> String {
+      let value = sourcekitd_variant_array_get_string(arr, index)!
+      return String(cString: value)
+    }
+
+    public func getInt(_ index: Int) -> Int {
+      let value = sourcekitd_variant_array_get_int64(arr, index)
+      return Int(value)
+    }
+
+    public func getBool(_ index: Int) -> Bool {
+      let value = sourcekitd_variant_array_get_bool(arr, index)
+      return value
+    }
+
+    public func getUID(_ index: Int) -> SourceKitdUID {
+      let value = sourcekitd_variant_array_get_uid(arr, index)!
+      return SourceKitdUID(uid: value)
+    }
+
+    public func getArray(_ index: Int) -> Array {
+      let value = sourcekitd_variant_array_get_value(arr, index)
+      return Array(arr: value)
+    }
+
+    public func getDictionary(_ index: Int) -> Dictionary {
+      let value = sourcekitd_variant_array_get_value(arr, index)
+      return Dictionary(dict: value)
+    }
+
+    public func enumerate(_ applier: (_ index: Int, _ value: Variant) -> Bool) {
+      // The block passed to sourcekit_variant_array_apply() does not actually
+      // escape, it's synchronous and not called after returning.
+      withoutActuallyEscaping(applier) { escapingApplier in
+        _ = sourcekitd_variant_array_apply(arr) { (index, elem) -> Bool in
+          return escapingApplier(Int(index), Variant(val: elem))
+        }
+      }
+    }
+
+    public var description: String {
+      return arr.description
+    }
+
+  }
+
+  public struct Variant: CustomStringConvertible {
+    private let val: sourcekitd_variant_t
+
+    fileprivate init(val: sourcekitd_variant_t) {
+      self.val = val
+    }
+
+    public func getString() -> String {
+      let value = sourcekitd_variant_string_get_ptr(val)!
+      let length = sourcekitd_variant_string_get_length(val)
+      return fromCStringLen(value, length: length)!
+    }
+
+    public func getStringBuffer() -> UnsafeBufferPointer<Int8> {
+      return UnsafeBufferPointer(start: sourcekitd_variant_string_get_ptr(val),
+                                 count: sourcekitd_variant_string_get_length(val))
+    }
+
+    public func getInt() -> Int {
+      let value = sourcekitd_variant_int64_get_value(val)
+      return Int(value)
+    }
+
+    public func getBool() -> Bool {
+      let value = sourcekitd_variant_bool_get_value(val)
+      return value
+    }
+
+    public func getUID() -> SourceKitdUID {
+      let value = sourcekitd_variant_uid_get_value(val)!
+      return SourceKitdUID(uid:value)
+    }
+
+    public func getArray() -> Array {
+      return Array(arr: val)
+    }
+
+    public func getDictionary() -> Dictionary {
+      return Dictionary(dict: val)
+    }
+
+    public func asAnyObject() -> AnyObject? {
```

Comment:
What's the reason you're using `AnyObject` and Foundation types here instead of stdlib types?

Reply:
oh, I'm not sure because this is borrowed code from internal clients. I'll try to convert to stdlib types.